pennylane/devices/default_qubit.py

```diff
@@ -206,6 +206,10 @@ def _apply_state_vector(self, state, device_wires):
             self._state = self._reshape(state, [2] * self.num_wires)
             return
 
+        # convert to the complex type used by the device
+        # the normalization inaccuracy might increase, do it after the checks
+        state = self._asarray(state, dtype=self.C_DTYPE)
```

Comment:
As suggested by the comment, moved this after the check, otherwise came across errors such as (last two values being the output of the norm and 1):
```python
pennylane/devices/default_qubit.py:129: in _apply_operation
    self._apply_state_vector(operation.parameters[0], wires)
pennylane/devices/default_qubit.py:201: in _apply_state_vector
    raise ValueError("Sum of amplitudes-squared does not equal one.")
E   ValueError: Sum of amplitudes-squared does not equal one.
-------------------------------------------------------------- Captured stdout call --------------------------------------------------------------
1.000000067179426 1.0
```
This would bring the question if we'd like to renormalize after calling `_asarray`. Imo we could leave it as is or if we'd go with renormalization, then we'd have to add it at other places too ([1](https://github.com/PennyLaneAI/pennylane/blob/15cc4fe5b08e01eea2e4e8d6e40fe4cf28d6ccb9/pennylane/devices/default_qubit.py#L173), [2](https://github.com/PennyLaneAI/pennylane/blob/15cc4fe5b08e01eea2e4e8d6e40fe4cf28d6ccb9/pennylane/devices/default_qubit.py#L221)).

Reply:
+1 for not normalizing here